extensions/amp-social-share/0.1/amp-social-share.js

```diff
@@ -0,0 +1,115 @@
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {getSocialConfig} from './amp-social-share-config';
+import {addParamsToUrl} from '../../../src/url.js';
+import {closest} from '../../../src/dom.js';
+
+const DEFAULT_WIDTH = 60;
+const DEFAULT_HEIGHT = 44;
+
+const CONTAINER_TAG = 'span';
+const LINK_TAG = 'a';
+
+class AmpSocialShare extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    this.config_ = this.getElementConfig(this.element);
+    AMP.assert(this.config_, 'The <script> config is required for ' +
+      '<amp-social-share> %s');
+
+    this.type_ = AMP.assert(this.element.getAttribute('type'),
+        'The type attribute is required for <amp-social-share> %s',
+        this.element);
+    this.typeConfig_ = getSocialConfig(this.type_);
+
+    this.container_ = this.renderShare(this.config_);
+
+    this.width = this.element.getAttribute('width') || DEFAULT_WIDTH;
+    this.height = this.element.getAttribute('height') || DEFAULT_HEIGHT;
+
+    this.element.appendChild(this.container);
+  }
+
+  /** @override */
+  layoutCallback() {
+    return Promise.resolve();
+  }
+
+  renderShare(config) {
+    const urlParams = {};
+
+    for (const param in this.typeConfig_['params']) {
+      const paramConf = this.typeConfig_['params'][param];
+      AMP.assert(!paramConf['required'] || param in config,
+        param + ' is a required configuration item for ' + this.type_);
+
+      if (param in config) {
+        const value = config[param];
+        if ('maxlength' in paramConf) {
+          const maxlength = paramConf['maxlength'];
+          AMP.assert(value.length < maxlength,
+            param + ' cannot exceed ' + maxlength);
+        }
+        urlParams[paramConf['param']] = value;
+      } else if (paramConf['type'] == 'fixed') {
+        urlParams[paramConf['param']] = paramConf['value'];
+      }
+    }
+    let container = closest(this.element, el => el.tagName == CONTAINER_TAG);
+    let link = closest(this.element, el => el.tagName == LINK_TAG);
+    if (container === null) {
+      container = this.getWin().document.createElement(CONTAINER_TAG);
+      container.classList.addClass('-amp-social-share');
+      container.classList.addClass('-amp-social-share-' + this.type_);
+    }
+    if (link === null) {
+      link = this.getWin().document.createElement(LINK_TAG);
+      link.innerHTML = this.typeConfig_['text'];
+    }
+    container.appendChild(link);
+    if (link.getAttribute('href') === null) {
+      link.setAttribute('href', addParamsToUrl(this.typeConfig_['url'],
+        urlParams));
+    }
+    if (link.getAttribute('target') === null) {
+      link.setAttribute('target', '_blank');
+    }
+
+    return container;
+  }
+
+  getElementConfig(element) {
+    const script = this.getFirstSubElement(element, 'script');
```

Comment:
i dont think this exists anymore right?

<img width="663" alt="screen shot 2016-02-11 at 1 29 38 am" src="https://cloud.githubusercontent.com/assets/354746/12973073/fb8fe708-d05e-11e5-9c1f-fd0e198279a9.png">